manifest: net: openthread: Regular openthread upmerge to `e10a925`

maciejbaczmanski opened the pull request:
Regular openthread upmerge to `e10a92570f94ff1e0bc5e0da9ecf0ee135d955a6`.

Additionally commit aligns platform implementation with diag changes:
https://github.com/openthread/openthread/pull/10354

based on:
https://github.com/openthread/ot-nrf528xx/pull/822

maciejbaczmanski committed:
manifest: net: openthread: Regular openthread upmerge to `e10a925`.

Regular openthread upmerge to `e10a92570f94ff1e0bc5e0da9ecf0ee135d955a6`.

Additionally commit aligns platform implementation with diag changes:
https://github.com/openthread/openthread/pull/10354

based on:
https://github.com/openthread/ot-nrf528xx/pull/822

Signed-off-by: Maciej Baczmanski <maciej.baczmanski@nordicsemi.no>

rlubos reviewed (dismissed):
Looks good, but we're in a code freeze now so it needs to wait.

canisLupus1313 commented:
@maciejbaczmanski You have created two commits in this PR. However, the changes must be uniform for each revision to compile correctly.

maciejbaczmanski commented:
> @maciejbaczmanski You have created two commits in this PR. However, the changes must be uniform for each revision to compile correctly.

thanks for spotting that. squashed commits and updated commit message
